compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -792,6 +792,10 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     );
                 }
             }
+            ObligationCauseCode::LetElse => {
+                err.help("try adding a `return` or `panic!(..)`");
```

Comment:
```suggestion
                err.help("try adding a diverging expression, such as `return` or `panic!(..)`");
```

May be more clear?

Reply:
Yes, I think that will be more helpful if the user doesn't know what "diverging" means. 👍